UnityProject/Assets/Scripts/Systems/Antagonists/Antags/Changeling/ChangelingAbility/ChangelingAbility.cs

```diff
@@ -0,0 +1,839 @@
+					target.playerHealth.reagentPoolSystem.BloodPool.RemoveVolume(toRemove);
+			}
+		}
+
+		private bool UseAbilityToggle(ChangelingMain changeling, ChangelingData data, bool toggle)
+		{
+			if (!data.IsToggle)
+				return false;
+			switch (data.abilityType)
+			{
+				case ChangelingAbilityType.Sting:
+					break;
+				case ChangelingAbilityType.Heal:
+					switch (data.healType)
+					{
+						case ChangelingHealType.RevivingStasis:
+							if (toggle)
+							{
+								UIActionManager.SetServerSpriteSO(this, ActionData.Sprites[1]);
+								changeling.isFakingDeath = true;
+
+								changeling.ChangelingMind.Body.playerHealth.StopMetabolismAndHeart();
+								changeling.ChangelingMind.Body.playerHealth.StopOverralCalculation();
+								changeling.ChangelingMind.Body.playerHealth.SetConsciousState(ConsciousState.UNCONSCIOUS);
+							} else
+							{
+								UIActionManager.SetServerSpriteSO(this, ActionData.Sprites[0]);
+								changeling.UseAbility(this);
+								// healing
+								changeling.ChangelingMind.Body.playerHealth.FullyHeal();
+								changeling.ChangelingMind.Body.playerHealth.UnstopOverallCalculation();
+								changeling.ChangelingMind.Body.playerHealth.UnstopMetabolismAndRestartHeart();
+								changeling.isFakingDeath = false;
+							}
+							return true;
+					}
+					break;
+				case ChangelingAbilityType.Misc:
+					switch (data.miscType)
+					{
+						case ChangelingMiscType.AugmentedEyesight:
+							if (Camera.main == null ||
+								Camera.main.TryGetComponent<CameraEffectControlScript>(out var effects) == false) return false;
+
+							foreach (var bodyPart in changeling.ChangelingMind.Body.playerHealth.BodyPartList)
+							{
+								foreach (BodyPartFunctionality organ in bodyPart.OrganList)
+								{
+									if (organ is Eye eye)
+									{
+										eye.ApplyChangesXrayState(toggle);
+										//goto SkipingFor;
+									}
+								}
+							}
+
+							//SkipingFor:
+							effects.AdjustPlayerVisibility(
+								toggle ? AbilityData.ExpandedNightVisionVisibility : effects.MinimalVisibilityScale,
+								toggle ? AbilityData.DefaultvisibilityAnimationSpeed : AbilityData.RevertvisibilityAnimationSpeed);
+							effects.ToggleNightVisionEffectState(toggle);
+							effects.SetNightVisionMaxLensRadius(true);
+
+							foreach (var x in changeling.AbilitiesNow)
+							{
+								if (x.AbilityData == AbilityData)
+								{
+									x.isToggled = toggle;
+								}
+							}
+							return true;
+					}
+					break;
+			}
+			return false;
+		}
+
+		private bool UseAbilityWithParam(ChangelingMain changeling, ChangelingData data, List<string> param)
+		{
+			switch (data.abilityType)
+			{
+				case ChangelingAbilityType.Sting:
+					break;
+				case ChangelingAbilityType.Heal:
+					break;
+				case ChangelingAbilityType.Transform:
```

Comment:
Wouldn't it be better if you split each ability into their own script?

Reply:
Think it`s better idea to just split them into methods